src/service/resource.js

```diff
@@ -365,19 +390,29 @@ export class Resource {
    * transitioned to the "ready for layout" state.
    */
   measure() {
+    let current = this;
+    let ancestor = this.getAmpAncestor();
     // Check if the element is ready to be measured.
-    // Placeholders are special. They are technically "owned" by parent AMP
-    // elements, sized by parents, but laid out independently. This means
-    // that placeholders need to at least wait until the parent element
-    // has been stubbed. We can tell whether the parent has been stubbed
-    // by whether a resource has been attached to it.
-    if (this.isPlaceholder_ &&
-        this.element.parentElement &&
-        // Use prefix to recognize AMP element. This is necessary because stub
-        // may not be attached yet.
-        startsWith(this.element.parentElement.tagName, 'AMP-') &&
-        !(RESOURCE_PROP_ in this.element.parentElement)) {
-      return;
+    while (ancestor) {
+      const resource = Resource.forElementOptional(ancestor);
+      // If there's an AMP ancestor that's still not stubbed (has a Resource),
+      // we don't know what to do yet.
+      if (!resource) {
```

Comment:
I.e. https://github.com/ampproject/amphtml/blob/4435faeedc6585acd3bf6758e746b0fa388eb257/extensions/amp-sticky-ad/1.0/amp-sticky-ad.js#L96

This code is probably benign, since no one ads stuff into this element from what I can tell. But others might not be so lucky.